src/components/JaegerIntegration/JaegerScatter.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { ChartWithLegend } from '@kiali/k-charted-pf4';
+import { ChartScatter } from '@patternfly/react-charts';
+import { JaegerTrace } from '../../types/JaegerInfo';
+import { isErrorTag } from './RouteHelper';
+import { PfColors } from '../Pf/PfColors';
+
+interface JaegerScatterProps {
+  traces: JaegerTrace[];
+  onClick: (idTrace) => void;
+  errorTraces?: boolean;
+}
+
+const ONE_MILLISECOND = 1000000;
+
+export class JaegerScatter extends React.Component<JaegerScatterProps> {
+  getColor(trace: JaegerTrace) {
+    const numErredSpans = trace.spans.filter(sp => sp.tags.some(isErrorTag)).length;
+    if (numErredSpans > 0) {
+      return PfColors.Red200;
+    }
+    return PfColors.Blue200;
+  }
+
+  render() {
+    const datapoints: any[] = [];
+    let seed = 1575987817125390;
+    for (let _i = 0; _i < 50; _i++) {
+      datapoints.push({
+        x: seed,
+        y: Math.random(),
+        name: `Trace ${_i}`,
+        unit: 'seconds',
+        size: 8
+      });
+      seed += 1000000;
+    }
+
```

Comment:
Hi @jotak maybe you can help me here, I used the same sample that we have in the storybook https://github.com/kiali/k-charted/pull/66 (It's working in onClick method)

Reply:
sorry I missed this comment.
I suspect that the time conversion is wrong, either here, or when you fetch data. Because even when I select something like "last 30 minutes", I only see plots for the last minute.

Reply:
Ok, I finally get it, and I better understand why @lucasponce was seeing weird x-axis plots in this comment: https://github.com/kiali/kiali-ui/pull/1577#issuecomment-565505249

This happens when there are many traces produced, because we're limiting to a small number (20 by default) so only the 20 last traces are shown. So, say we have 10 traces per second and we ask for traces in the last 30 minutes, the X axis will show only the 2 last seconds, making it difficult to understand.

I would say we have 2 problems here:
- the chart range adapts automatically to the data it contains, regardless the queried time frame. So we ask for 30 minutes but it can show much less than that, due to limiting. There's already an issue opened in k-charted about that: https://github.com/kiali/k-charted/issues/15 ; we could do that.

- when there's many traces, the query will return only the last ones. I don't know how this is worked around in Jaeger UI, but it doesn't seem to happen there: it looks like the traces are randomly ordered and not ordered by date. In the metrics page with spans overlay, I tried to reduce this problem by showing only spans above some percentile, but it's still imperfect.

Reply:
I think this one is addressed in the lastest commits:
![image](https://user-images.githubusercontent.com/1662329/72429166-96ea2600-378f-11ea-9a04-cef0ab57364c.png)
![image](https://user-images.githubusercontent.com/1662329/72429182-9f426100-378f-11ea-8671-9acc0dc7314e.png)

Due nature of the query of Jaeger traces (where you can select "give me last 20 traces from last 6h" and those "20 traces can be in the first minutes").

There is now an "adjust time" checkbox to see this in the x-asis with the interval asked (i.e. 6h) or just to adjust the x-axis with the size of the query (i.e. last 20 traces).

It's more an experimental way to handle this, I think it would allow user to put in context the results.